schema/query/projection_evaluator.go

```diff
@@ -50,23 +53,31 @@ func evalProjection(ctx context.Context, p Projection, payload map[string]interf
 						return nil, fmt.Errorf("%s: invalid value: not a dict", pf.Name)
 					}
 					var err error
-					if subval, err = evalProjection(ctx, pf.Children, subval, def.Schema, resolver); err != nil {
+					if subval, err = evalProjection(ctx, pf.Children, subval, def.Schema, rbr); err != nil {
 						return nil, fmt.Errorf("%s.%v", pf.Name, err)
 					}
 					if res[name], err = resolveFieldHandler(ctx, pf, def, subval); err != nil {
 						return nil, err
 					}
 				} else if ref, ok := def.Validator.(*schema.Reference); ok {
-					// Do not execute the sub-request right away, store a
-					// asyncProjection type of lambda that will be executed later
-					// with concurrency control
-					res[name] = asyncGet{
-						payload: res,
-						field:   pf,
-						path:    ref.Path,
-						id:      val,
-						def:     def,
-					}
+					// Execute sub-request in batch
+					q := &Query{Predicate: Predicate{Equal{Field: "id", Value: val}}}
+					rbr.request(ref.Path, q, func(payloads []map[string]interface{}, validator schema.Validator) error {
+						resMu.Lock()
+						defer resMu.Unlock()
```

Comment:
should resMu really just be locked while you read out res[name]? Does it make sense to make it a RWMutex?